renderdoc/driver/vulkan/vk_resources.h

```diff
@@ -1055,19 +1099,16 @@ struct ImgRefs
   int GetAspectCount() const;
 
   ImgRefs() : initializedLiveRes(NULL) {}
-  inline ImgRefs(VkImageType type, VkImageAspectFlags aspectMask, int levelCount, int layerCount,
-                 VkExtent3D extent)
+  inline ImgRefs(const ImageInfo &imageInfo)
       : rangeRefs(1, eFrameRef_None),
-        initializedLiveRes(NULL),
-        type(type),
-        aspectMask(aspectMask),
-        levelCount(levelCount),
-        layerCount(layerCount),
-        extent(extent)
+        imageInfo(imageInfo),
+        aspectMask(FormatImageAspects(imageInfo.format))
   {
-    if(type == VK_IMAGE_TYPE_3D)
+    if(imageInfo.imageType == VK_IMAGE_TYPE_3D)
       // Depth slices of 3D views are treated as array layers
-      layerCount = extent.depth;
+      this->imageInfo.layerCount = imageInfo.extent.depth;
+    else
+      this->imageInfo.layerCount = imageInfo.layerCount;
```

Comment:
Is this else needed? It seems redundant after having copied imageInfo already.

Reply:
Removed.